src/sentry/static/sentry/app/views/organizationStream/overview.jsx

```diff
@@ -594,21 +601,26 @@ const OrganizationStream = createReactClass({
     let hasReleases = false;
     let projectId = null;
     let latestRelease = null;
+
     const {selectedProject} = this.state;
+    const projects = this.getGlobalSearchProjects();
+
     if (selectedProject) {
       const features = new Set(selectedProject.features);
       hasReleases = features.has('releases');
       latestRelease = selectedProject.latestRelease;
       projectId = selectedProject.slug;
-    } else {
+    } else if (projects.length == 1) {
       // If the user has filtered down to a single project
       // we can hint the autocomplete/savedsearch picker with that.
-      const projects = this.getGlobalSearchProjects();
-      if (projects.length === 1) {
-        projectId = projects[0].slug;
-      }
+      projectId = projects[0].slug;
     }
 
+    const tagValueLoader = (key, search) => {
```

Comment:
Can we make this a method? I'm not sure how optimized `StreamFilters` and `StreamSidebar` are, but this could affect prop comparisons in lifecycle methods since we're creating a new function on each render.